### PR TITLE
feat: add 'extra' to eventOptions

### PR DESCRIPTION
**Pull request by falconandy**
### Summary

Adds `extra` to eventOptions.

### Checklist

* [X] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-TypeScript/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  no

### CODE REVIEW

File: packages/analytics-core/test/timeline.test.ts [hunk 1]
```diff
@@ -128,6 +128,62 @@ describe('timeline', () => {
       await timeline.deregister(before.name);
       expect(beforeExecute).toHaveBeenCalledTimes(0);
     });
+
+    test("should not pass event's extra to destination plugins", async () => {
```
Comment (justin-fiedler) [inline packages/analytics-core/test/timeline.test.ts:132]: We want to pass the extra to the destination plugins to allow users to access it, but do not want to include it in the payload sent to the server in the Amplitude Destination.

Comment (justin-fiedler) [reply packages/analytics-core/test/timeline.test.ts:132]: @kevinpagtakhan I noticed that the Amplitude Destination Plugin is named `Destination` do you think we could rename it to be more explicit e.g. `AmplitudeDestinationPlugin`?

**Review by justin-fiedler — CHANGES_REQUESTED** [review]

File: packages/analytics-core/test/timeline.test.ts (hunk 1 above)
Comment (kevinpagtakhan) [reply packages/analytics-core/test/timeline.test.ts:132]: Sure, that makes sense!

Comment (falconandy) [reply packages/analytics-core/test/timeline.test.ts]: > We want to pass the extra to the destination plugins to allow users to access it

Fixed

>  noticed that the Amplitude Destination Plugin is named Destination do you think we could rename it to be more explicit e.g. AmplitudeDestinationPlugin?

In this PR?

Comment (justin-fiedler) [reply packages/analytics-core/test/timeline.test.ts]: Thanks @falconandy we can rename the plugin in another PR. I just wanted to call it out.

**Review by justin-fiedler — APPROVED** [review]

**Pull request merged**